server/routes/auth/dropbox.js

```diff
@@ -0,0 +1,56 @@
+const router = require('express').Router()
+const request = require('request')
+const { DROPBOX_CLIENT_ID, DROPBOX_CLIENT_SECRET } = require('../../../secrets')
+const { User } = require('../../db')
+module.exports = router
+
+router.get('/callback', function(req, res) {
+  if (!req.query || !req.query.state) return res.status(400).send('No user')
+  if (!req.query.code) return res.status(400).send('No code')
+  const url = 'https://api.dropbox.com/oauth2/token'
+
+  request.post(
+    url,
+    {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization:
+          'Basic ' +
+          Buffer.from(DROPBOX_CLIENT_ID + ':' + DROPBOX_CLIENT_SECRET).toString(
+            'base64'
+          )
+      },
+      form: {
+        code: req.query.code,
+        grant_type: 'authorization_code',
+        redirect_uri: 'http://localhost:3001/api/v0/auth/dropbox/callback'
```

Comment:
Hey @listenaddress could you explain how this is working? It seems like you're already inside this route that's getting passed as the `redirect_uri` param, so once the request is sent, don't we just end up back in this route? I feel like this would infinitely loop?